datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+    -------
+    """
+    # If no colors are given, all substructures are depicted in gray.
+    if color_list is None:
+        color_list = [(0.5, 0.5, 0.5)] * len(indices_set_lists)
+    if len(color_list) < len(indices_set_lists):
+        raise ValueError("Not enough colors for referenced substructures!")
+
+    level_manager = defaultdict(set)
+    for match_atoms, color in zip(indices_set_lists, color_list):
+        used_levels = set.union(*[level_manager[a] for a in match_atoms])
+        if len(used_levels) == 0:
+            free_levels = {0}
+        else:
+            max_level = max(used_levels)
+            free_levels = set(range(max_level)) - used_levels
+
+        if free_levels:
+            draw_level = min(free_levels)
+        else:
+            draw_level = max(used_levels) + 1
+
+        for a in match_atoms:
+            level_manager[a].add(draw_level)
+
+        ar = r_min + r_dist * draw_level
+        _draw_substructurematch(
+            canvas,
+            mol,
+            match_atoms,
+            rel_radius=ar,
+            rel_width=max(relative_bond_width, ar),
+            color=color,
+            line_width=line_width,
+        )
+
+
+color_dict = {
+    "red": (1, 0, 0, 1),
+    "blue": (0, 0.5, 1, 1),
+    "orange": (1, 0.5, 0, 1),
+    "green": (0, 1, 0, 1),
+    "yellow": (1, 1, 0, 1),
+    "dark blue": (0, 0, 0.5, 1),
+}
+
+
+def lasso_highlight_image(
+    canvas_width: int,
+    canvas_height: int,
+    target_molecule: Union[str, dm.Mol],
+    search_molecules: Union[str, List[str], dm.Mol, List[dm.Mol]],
+) -> Image:
+    """A generalized interface to access both highlighting options whether the input is as a smiles, smarts or mol
+
+    Args:
+        canvas_width (int): _description_
+        canvas_height (int): _description_
+        target_molecule (Union[str, dm.Mol]):
+        search_molecules (Union[str, List[str], dm.Mol, List[dm.Mol]]): Various ways to enter but the search molecules must be entered as smart
+    """
+
+    # check if the input is valid
+    if target_molecule is None or (isinstance(target_molecule, str) and len(target_molecule) == 0):
+        raise ValueError("Please enter a valid target molecule or smiles")
+
+    if search_molecules is None or (
+        isinstance(search_molecules, str) and len(search_molecules) == 0
+    ):
+        raise ValueError("Please enter valid search molecules or smarts")
+
+    # less than 1 throws File parsing error: PNG header not recognized over 5,000 leads to a DecompressionBombError later on
+    if canvas_width < 1 or canvas_width > 5000 or canvas_height < 1 or canvas_height > 5000:
+        raise ValueError(
+            "To avoid errors please choose a number between 1-5000 for the canvas width or height"
+        )
+
+    if isinstance(target_molecule, str):
+        target_molecule = dm.to_mol(target_molecule)
+
+    mol = prepare_mol_for_drawing(target_molecule, kekulize=True)
+    d = rdMolDraw2D.MolDraw2DCairo(500, 500)
```

Comment:
As said above, it's important to also support SVG as output. I propose not to depend on an external library (anyway PNG -> SVG is not really possible, only the reverse is possible).

Instead we should simply follow what rdkit does, which is to simply put a boolean flag and init the correct rdkit drawer automatically under the hood. See https://github.com/rdkit/rdkit/blob/7da2eb011a20785d2ad4aa46b1dfc37675396c6e/rdkit/Chem/Draw/__init__.py#L607-L624 for details.

Then it would be nice to use the same behaviour as `dm.to_image()` using the same boolean flag and default value: `use_svg: bool = True,`.

Reply:
Thank you, @hadim and @cwognum. I have it set up correctly for this, I believe, although I am struggling right now.  When I try to match the png workflow, it doesn't work. It returns the SVG code. Shown below is what it looks like.  

```<?xml version='1.0' encoding='iso-8859-1'?>\n<svg version='1.1' baseProfile='full'\n              xmlns='http://www.w3.org/2000/svg'\n                      xmlns:rdkit='http://www.rdkit.org/xml'\n                      xmlns:xlink='http://www.w3.org/1999/xlink'\n                  xml:space='preserve'\nwidth='400px' height='400px' viewBox='0 0 400 400'>\n<!-- END OF HEADER -->\n<rect style='opacity:1.0;fill:#FFFFFF;stroke:none' width='400.0' height='400.0' x='0.0' y='0.0'> </rect>\n<path class='bond-0 atom-0 atom-1' d='```

I'm not sure how this was turned into the actual image in to_image, but I was hoping I could get some help with this last part. I have shown how I have been able to get it working in Visualization.ipynb from my most recent commit.

Reply:
@dessygil, this is expected behaviour.  `to_image` uses `Draw.MolsToGridImage`  which rdkit automatically patches internally to return ipython SVG objects:

### Code that modify behaviour here

https://github.com/rdkit/rdkit/blob/af531a79cadc96549a06fda1cd4bce282bf4f22b/rdkit/Chem/Draw/IPythonConsole.py#L374

### Automatic patching during rdkit import here (which will call `InstallIPythonRenderer()`) 

https://github.com/rdkit/rdkit/blob/af531a79cadc96549a06fda1cd4bce282bf4f22b/rdkit/__init__.py#L23 


You can take inspiration from this snippet of the code above:

https://github.com/rdkit/rdkit/blob/af531a79cadc96549a06fda1cd4bce282bf4f22b/rdkit/Chem/Draw/IPythonConsole.py#L271-L275

Basically:

1. Check if we are in a jupyter notebook
2. If yes,  wrap the SVG string into a `IPython.display.SVG`

Not sure if you need to do anything for PNG (if PIL image), but you can give it a try.